sway-fmt-v2/src/items/item_storage.rs

```diff
@@ -1,15 +1,139 @@
+
+            if current_variant_length < max_valid_variant_length {
+                // We need to add alignment between : and ty
+                // max_valid_variant_length: the length of the variant that we are taking as a reference to allign
+                // current_variant_length: the length of the current variant that we are trying to format
+                let required_alignment = max_valid_variant_length - current_variant_length;
+                formatted_code.push_str(&(0..required_alignment).map(|_| ' ').collect::<String>());
+            }
+            // Push the colon token
+            formatted_code.push_str(item.colon_token.span().as_str());
+            formatted_code.push(' ');
+
+            // Push the ty
+            formatted_code.push_str(item.ty.span().as_str());
+
+            // Push initializer if it exists.
+            if let Some(initializer) = item.initializer {
+                // Push a ` `
+                formatted_code.push(' ');
+
+                let expr = initializer.1;
+
+                // Push the `=`
+                formatted_code.push(PunctKind::Equals.as_char());
+
+                // Push a ` `
+                formatted_code.push(' ');
+
+                // Push the unformatted expr
+                formatted_code.push_str(expr.span().as_str());
+            }
+
+            // TODO we are currently pushing \n directly, if we want to format storage
+            // into a single line in some cases. We should handle this better!
+            formatted_code.push(PunctKind::Comma.as_char());
+            formatted_code.push('\n');
+        }
+
+        // Add `}`
+        Self::close_curly_brace(formatted_code, formatter)?;
+
+        Ok(())
+    }
+}
+
+impl CurlyBrace for ItemStorage {
+    fn open_curly_brace(
+        line: &mut String,
+        formatter: &mut Formatter,
+    ) -> Result<(), FormatterError> {
+        let brace_style = formatter.config.items.item_brace_style;
+        let extra_width = formatter.config.whitespace.tab_spaces;
+        let mut shape = formatter.shape;
+        let open_brace = Delimiter::Brace.as_open_char();
+        match brace_style {
+            ItemBraceStyle::AlwaysNextLine => {
+                // Add openning brace to the next line.
+                line.push_str(&format!("\n{}\n", open_brace));
+                shape = shape.block_indent(extra_width);
+            }
+            _ => {
+                // Add opening brace to the same line
+                line.push_str(&format!(" {}\n", open_brace));
+                shape = shape.block_indent(extra_width);
+            }
+        }
+
+        formatter.shape = shape;
+        Ok(())
+    }
+    fn close_curly_brace(
+        line: &mut String,
+        formatter: &mut Formatter,
     ) -> Result<(), FormatterError> {
-        todo!()
+        line.push(Delimiter::Brace.as_close_char());
+        // If shape is becoming left-most alligned or - indent just have the defualt shape
```

Comment:
I don't quite understand this comment here

Reply:
I refactored the comment here 👍 Basically `shrink_left` will return an error and if the current indentation level is becoming invalid, (i.e, current indentation level is becoming `< 0`)  in that case we should use the `Shape::default` which is basically `indentation level = 0`.